File: 380/solution.py
import random


class RandomizedSet:

    def __init__(self):
        self.nums = []
        self.data = {}

    def insert(self, val: int) -> bool:
        if val in self.data:
            return False
        self.data[val] = len(self.nums)
        self.nums.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val not in self.data:
            return False
        del_idx = self.data[val]
        last_idx = len(self.nums) - 1
        self.data[self.nums[last_idx]] = del_idx
        self.nums[del_idx] = self.nums[last_idx]
        del self.data[val]
        self.nums.pop()
        return True

    def getRandom(self) -> int:
        return random.choice(self.nums)



# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 380/Solution.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Random;

class RandomizedSet {
    private List<Integer> nums;
    private HashMap<Integer, Integer> data;

    public RandomizedSet() {
        nums = new ArrayList<>();
        data = new HashMap<>();
    }
    
    public boolean insert(int val) {
        if (data.containsKey(val))
            return false;
        data.put(val, nums.size());
        nums.add(val);
        return true;
    }
    
    public boolean remove(int val) {
        if (!data.containsKey(val))
            return false;
        int delIdx = data.get(val);
        int lastIdx = nums.size() - 1;
        nums.set(delIdx, nums.get(lastIdx));
        data.put(nums.get(lastIdx), delIdx);
        nums.remove(lastIdx);
        data.remove(val);
        return true;
    }
    
    public int getRandom() {
        int idx = new Random().nextInt(nums.size());
        return nums.get(idx);
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */